### ISSUE TITLE
Allow sorting and/or paging of output

### ISSUE THREAD

**Issue by rartino** [Materials-Consortia/OPTiMaDe#20]
Allow a sort query parameter, e.g., something like:
```
    ?filter=element_ratios HAS element = "Si"&order=chemical_formula
```
Ascending, descending? Multiple fields?

**Comment by rartino** [comment]
So, how about the syntax:
```
 ?filter=element_ratios HAS element = "Si"&order=chemical_formula
 ?filter=element_ratios HAS element = "Si"&order=chemical_formula:asc
 ?filter=element_ratios HAS element = "Si"&order=chemical_formula:desc

 ?filter=element_ratios HAS element = "Si"&order=chemical_formula:asc,elements:dsec
```
Where the first defaults to desc = descending.

**Comment by rartino** [comment]
Approved as optional

**Comment by merkys** [comment]
Suggestion for discovery mechanism: at the entry introspection endpoints viz. ``/structures/info`` we have the following data structure:
```
{
  "properties" :
    {
      "some_property" : { "description" : "free text description", "unit" : "unit description" }
    }
}
```
We may include additional attribute ``sortable`` alongside ``description`` and ``unit``. Value of this attribute would be boolean, and ``False`` by default. Therefore, databases that choose to implement sorting would need to explicitly indicate so for all sortable properties.

**Comment by dwinston** [comment]
I suggest using the `page` key for pagination concerns as per [https://jsonapi.org/format/1.0/#fetching-paginationI](https://jsonapi.org/format/1.0/#fetching-pagination). Specifically, I suggest replacing the `response_limit` key in our spec with `page[limit]` and using its counterpart `page[offset]` in order to fully specify pagination as an offset-based strategy. The alternative suggested by JSON API 1.0 is `page[number]` and `page[size]` for a page-based strategy. My preference is for an offset-based strategy. Thus, while paging is OPTIONAL, I would us to specify the paging strategy and keys used if one implements paging.

**Comment by dwinston** [comment]
### Decisions on sorting
- Support for the `sort` query parameter for an entry listing endpoint is optional. If an implementation supports sortable queries, it MUST use the `sort` query parameter with format as specified by the [JSON API 1.0 spec](https://jsonapi.org/format/1.0/#fetching-sorting).
- It is not required that an implementation supports multiple sort fields for a single query. However, if it does, it again must conform to the [JSON API 1.0 spec](https://jsonapi.org/format/1.0/#fetching-sorting).
- If an implementation supports sorting for an entry listing endpoint, then the` /<entries>/info` endpoint MUST include, for each field name `<field>` in its `data.properties.<field>` response value, the key "sortable" with value `true`. This is in addition to the property description (and optional unit).
### Decisions on pagination
- Pagination is MANDATORY. Implementations MUST conform to the [JSON API 1.0 spec](https://jsonapi.org/format/1.0/#fetching-pagination). That is, when a server does not return all data matching a query (which it must be capable of, because a user may request a small `page[limit]` -- see below), it must provide a `links` object in the response that includes `next`, `prev`, `first`, and `last` values -- in particular, we require each of these keys to have either a valid URL or `null` as their value. However, there are only two situations where the value of a key is (and MUST be) `null`: in the first-page response, the value of `prev` is `null`, and in the last-page response, the value of `next` is `null`. In all other cases, the value of each of these keys must be a valid URL.
- Following recommendations of the JSON API 1.0 spec and its reservation of the `page` query argument for pagination, implementers MUST use an offset-based pagination strategy using the `page[offset]` and `page[limit]` query arguments. It is trivial for an implementation to map this strategy to a page-based strategy using arithmetic on the server side.
- If a server sets a maximum `page[limit]` and a user specifies a larger limit, the server MUST return an error. For example, if a server `page[limit]` max is 100, and a user specifies 200, the server cannot return 100 entries with a warning. The server MUST return an error.
- Thus, gien the above specification on pagination, each URL value of a `links` field will include the appropriate `page[offset]` and `page[limit]` values while reproducing all other query parameters of the original request.

**Comment by rartino** [comment]
Closed by #97
